paasta_tools/contrib/timeouts_metrics_prom.py

```diff
@@ -0,0 +1,60 @@
+#!/opt/venvs/paasta-tools/bin/python
+"""
+The goal is to walk through all smartstack.yaml in yelpsoa directories
+and convert timeout values to the prometheus format.
+
+For every upstream and path, if endpoint timeout is specified, we record
+per endpoint timeout value. If not and if timeout_server_ms is specified,
+we record timeout_server_ms value in path `default`. Otherwise, we record the default
+timeout_server_ms 1000.
+
+# HELP endpoint timeout value defined in yelpsoa-config.
+# TYPE TYPE yelpsoaconfig_endpoint_timeouts_ms gauge
+yelpsoaconfig_endpoint_timeouts_ms{path="/consumer_app/devices/braze/info",upstream="push_notifications.main.egress_cluster"} 10000.0
+yelpsoaconfig_endpoint_timeouts_ms{path="default",upstream="mysql_read_security.main.egress_cluster"} 100.0
+"""
+import os
+
+import yaml
+from prometheus_client import CollectorRegistry
+from prometheus_client import Gauge
+from prometheus_client import write_to_textfile
+
+SOA_DIR = "/nail/etc/services/"
+PROM_OUTPUT_FILE = "/nail/etc/services/.autotune_timeouts.prom"
+
+
+def read_and_write_timeouts_metrics(root, service, prom_metric):
+    with open(os.path.join(root, "smartstack.yaml")) as smartstack_file:
+        smartstack_yaml = yaml.safe_load(smartstack_file)
+    for instance_name, info in smartstack_yaml.items():
+        upstream = service + "." + instance_name + ".egress_cluster"
+        if "endpoint_timeouts" in info:
+            for path, endpoint_timeouts in info["endpoint_timeouts"].items():
+                prom_metric.labels(path, upstream).set(endpoint_timeouts)
+        else:
```

Comment:
I think we don't want this in an `else:` block: we want to record the default/base timeout whether or not there are overrides

Reply:
Uh can you explain why we want to always record the default/base timeout? My vision was to get current per endpoint timeout in grafana, I didn't see why we would need default timeout.

Reply:
the "default" timeout will apply to any path that isn't matched by a per-endpoint timeout.  So if a service has endpoints "/action1" and "/action2", and only a per-endpoint timeout for "/action2", then the default timeout will apply for "/action1"

Reply:
you are right, and actually that was my intention but for some reasons my brain was fuzzy and the code was not followed up 😢

Reply:
fixed that!